File: backend/ecommerce/src/main/java/com/bookstore/ecommerce/app/data/Book.java
package com.bookstore.ecommerce.app.data;

import lombok.Getter;

public class Book {
  @Getter private String id;
  @Getter private String title;
  @Getter private String subTitle;
  @Getter private String description;

  public Book() {
  }

  public Book(String id, String title, String subTitle, String description) {
    this.id = id;
    this.title = title;
    this.subTitle = subTitle;
    this.description = description;
  }
}

File: backend/store/app/domain/data/book.go
package data

import (
	"time"
)

type Book struct {
	Id            EntityId `gorm:"primaryKey"`
	GoogleBookId  string
	Title         string
	Subtitle      string
	Description   string
	Authors       []string
	Publisher     string
	PublishedDate time.Time
	AverageRating float32
	RatingsCount  int
	ThumbnailUrl  string
	PreviewUrl    string
	OnhandQty     int
	ReservedQty   int
}

func (b *Book) GetId() EntityId {
	return b.Id
}

func (b *Book) SetId(id EntityId) {
	b.Id = id
}

func (b *Book) Clone() *Book {
	return &Book{
		Id:            b.Id,
		GoogleBookId:  b.GoogleBookId,
		Title:         b.Title,
		Subtitle:      b.Subtitle,
		Description:   b.Description,
		Authors:       b.Authors,
		Publisher:     b.Publisher,
		PublishedDate: b.PublishedDate,
		AverageRating: b.AverageRating,
		RatingsCount:  b.RatingsCount,
		ThumbnailUrl:  b.ThumbnailUrl,
		PreviewUrl:    b.PreviewUrl,
		OnhandQty:     b.OnhandQty,
		ReservedQty:   b.ReservedQty,
	}
}
